### DISCUSSION TITLE
MicrobiomeStat functions not showing up after installation in R

### DISCUSSION THREAD

**Discussion by niall142** [cafferychen777/MicrobiomeStat#64] [Q&A] [answered]
Hi again Chen, thanks for the recommendation to use MicrobiomeStat! I've run into some trouble getting the package loaded however; I'm not seeing any of the mStat_ functions. The only functions from MicrobiomeStat that R is showing me are linda, linda.plot, and linda.wald.test. 

The package version is 1.2,  I'm up to date on R, and I installed the various dependencies. I tried installing from GitHub instead of CRAN, but I get this error message: 

``Error: Failed to install 'MicrobiomeStat' from GitHub:
  ! System command 'Rcmd.exe' failed
``

I've tried on two computers now and gotten the same result. Any suggestions? 

Thanks!
-Niall

**Comment by cafferychen777** [comment, marked as answer]
Dear Niall,

Thank you for your patience. I've reviewed the dependencies for MicrobiomeStat, and I'd like to provide you with a more detailed installation process to ensure all necessary components are in place. Please follow these steps:

1. First, ensure you have R version 3.5.0 or newer installed. You can check your R version by running:

   ```R
   R.version$version.string
   ```

2. Install and load the pacman package, which we'll use to streamline the installation of other packages:

   ```R
   if (!require("pacman")) install.packages("pacman")
   library(pacman)
   ```

3. Install the required and suggested dependencies:

   ```R
   p_load(rlang, tibble, ggplot2, matrixStats, lmerTest, foreach, modeest, vegan, dplyr, pheatmap, tidyr, ggh4x, ape, GUniFrac, scales, stringr, rmarkdown, knitr, pander, tinytex, ggrepel, parallel, ggprism, aplot, yaml, biomformat, Biostrings, ggtree, ggtreeExtra)
   ```

   Note: Some of these packages (like Biostrings) are from Bioconductor. If you encounter any issues, you may need to install them separately using:

   ```R
   if (!require("BiocManager", quietly = TRUE))
       install.packages("BiocManager")
   BiocManager::install(c("Biostrings", "biomformat"))
   ```

4. Now, install MicrobiomeStat from GitHub:

   ```R
   devtools::install_github("cafferychen777/MicrobiomeStat")
   ```

5. If the GitHub installation fails, try the manual installation method:
   - Download the ZIP file from https://github.com/cafferychen777/MicrobiomeStat
   - Unzip the file
   - Install from the local source:
     ```R
     install.packages("path/to/unzipped/MicrobiomeStat", repos = NULL, type = "source")
     ```
     (Replace "path/to/unzipped/MicrobiomeStat" with the actual path)

6. After installation, load the package and verify the functions are available:

   ```R
   library(MicrobiomeStat)
   ls("package:MicrobiomeStat")
   ```

7. If you don't see the mStat_ functions, restart your R session and try loading the package again.

If you still encounter issues after following these steps, please provide:
- The output of `sessionInfo()`
- Any error messages you receive
- Your operating system details

This information will help us troubleshoot any persistent problems more effectively.

Don't hesitate to reach out if you need further assistance or clarification on any of these steps.

Best regards,
Chen Yang

**Comment by niall142** [comment]
Hi Chen,

That got it, thanks very much!!

-Niall
